src/main/java/seedu/address/storage/Authentication.java

```diff
@@ -0,0 +1,202 @@
+        try {
+            //If ClientBook is locked, encrypt the zip file using the password.
+            if (this.password.isPresent()) {
+                dataZip = new ZipFile(this.getZipPath(), this.password.get().toCharArray());
+                ZipParameters parameters = new ZipParameters();
+                parameters.setEncryptFiles(true);
+                parameters.setEncryptionMethod(EncryptionMethod.AES);
+                parameters.setAesKeyStrength(AesKeyStrength.KEY_STRENGTH_256);
+                parameters.setCompressionLevel(CompressionLevel.NORMAL);
+                parameters.setCompressionMethod(CompressionMethod.DEFLATE);
+                dataZip.addFile(this.filePath.toString(), parameters);
+
+            //ClientBook is not locked, zip the data .json file without a password.
+            } else {
+                dataZip = new ZipFile(this.getZipPath());
+                dataZip.addFile(this.filePath.toString());
+            }
+        } catch (ZipException e) {
+            logger.info("Failed to compress data .json file into a zip file");
+            throw e;
+        }
+
+
+
+
+    }
+
+    /**
+     * Sets a hook to lock the data .json file in the zip file when the application closes.
+     */
+    public void setShutDownHook() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                this.lock();
+                this.deleteJson();
+            } catch (ZipException e) {
+                e.printStackTrace();
+            }
+        }));
+    }
+
+    public String getPassword() {
+        return this.password.orElse("");
+    }
+
+    public boolean isPasswordPresent() {
+        return this.password.isPresent();
+    }
+
+    public void setPassword(String password) {
+        this.password = Optional.of(password);
+    }
+
+    public void removePassword() {
+        this.password = Optional.empty();
+    }
+
+    private void deleteJson() {
+        File dataJson = new File(this.filePath.toString());
+        dataJson.delete();
+    }
+
+    private boolean attemptUnzip(String password) throws ZipException {
+        ZipFile dataZip = new ZipFile(this.getZipPath());
+        //First check if zip file encrypted and set the password to be used.
+        try {
+            if (dataZip.isEncrypted()) {
+                dataZip.setPassword(password.toCharArray());
+            }
+        } catch (ZipException e) {
+            logger.info("Error attempting to check zip file at: " + this.getZipPath());
+            throw e;
+        }
+
+        //Attempt to unzip the file using the provided password.
+        try {
+            dataZip.extractAll(this.getFolderPath());
+            return true;
+        } catch (ZipException e) {
+            logger.info("Error when unzipping file at: " + this.getZipPath());
+            return false;
+        }
+    }
+
+    private boolean attemptUnzip() throws ZipException {
+        return attemptUnzip("");
+    }
```

Comment:
Is there some reason you chose to overload the attemptUnzip() method with this? It seems to me that you could directly call attemptUnzip("") where you want to call this method.

Reply:
It was to make it clear during usage that there is no password involved as someone reading it may interpret as entering an empty password.

Reply:
I see, that makes sense. Perhaps it could be clearer if you leave a comment or rename the method to something like

Reply:
I agree, I have updated the method's name to improve clarity.